sbi/simulators/simutils.py

```diff
@@ -117,56 +117,59 @@ def simulate_in_batches(
     parameter_sample_fn: Callable,
     num_samples: int,
     simulation_batch_size: int,
-    x_dim: torch.Size,
 ) -> (torch.Tensor, torch.Tensor):
     """
-    Draw `num_samples` parameter sets and simulate them in batches of size `simulation_batch_size`.
+    Draw `num_samples` parameter sets and simulate them in batches of size
+     `simulation_batch_size`.
 
     Features: - allows to simulate in batches of arbitrary size
               - if `simulation_batch_size==-1`, all simulations are run at the same time
               - simulator output can be np.array or torch.Tensor
 
     Args:
-        simulator: simulator function.
-            If `simulation_batch_size == 1`: takes in parameters of shape (1, num_dim_parameters)
-                and outputs xs of shape (1, num_dim_x)
-            If `simulation_batch_size > 1`: takes in thetas of shape (simulation_batch_size, num_dim_parameters)
-                and outputs xs of shape (simulation_batch_size, num_dim_x)
+        simulator: simulator function that takes in parameters theta and outputs data x
+            If `simulation_batch_size == 1`: takes in parameters of shape
+             (1, num_dim_parameters) and outputs xs of shape (1, num_dim_x)
+            If `simulation_batch_size > 1`: takes in thetas of shape
+             (simulation_batch_size, num_dim_parameters) and outputs xs of shape
+             (simulation_batch_size, num_dim_x)
         parameter_sample_fn: Function to call for generating theta, e.g. prior sampling
         num_samples: Number of simulations to run
         simulation_batch_size: Number of simulations that are run within a single batch
-            If `simulation_batch_size == -1`, we run a batch with all simulations required,
-            i.e. `simulation_batch_size = num_samples`
-        x_dim: dimensionality of a single simulator output
+            If `simulation_batch_size == -1`, we run a batch with all simulations
+             required, i.e. `simulation_batch_size = num_samples`
```

Comment:
What do you mean by 'mixing of the batch'?

Reply:
@michaeldeistler I think Alvaro means the confusion of the concepts of `sample shape` and `batch shape`. I am confused about that as well, although I am quite sure that here we should call it a `batch` shape because we are simulating different (not iid) parameter sets.

Reply:
Regarding name conventions for the shapes: the [function attributes set to the simulator](https://github.com/mackelab/sbi/blob/74a1a0435711a7e6cc78206012a9242bc118f018/sbi/simulators/simutils.py#L38-L40) are named after the previous class properties: `parameter_dim` and `observation_dim`. 

We could put a `num_` in front, but we should always call it "parameter" and "observation" I find.